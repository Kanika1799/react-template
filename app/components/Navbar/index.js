/**
 *
 * Navbar
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage as T } from 'react-intl';
import { Col, Row } from 'antd';
import star from '@images/star.svg';
import icon from '@images/icon-notification.svg';

const MainDiv = styled.div`
  padding-top: 10px;
  .Dashboard {
    width: 188px;
    height: 20px;
    font-family: Inter;
    font-size: 20px;
    font-weight: 500;
    font-stretch: normal;
    font-style: normal;
    line-height: 0.67;
    letter-spacing: normal;
    text-align: center;
    margin-left: 5px;
    margin-top: 15px;
    color: #1a2940;
  }
  .ModifiedCol {
    margin-left: 1050px;
    margin-top: 20px;
  }
  .HeadingDiv {
    width: 1400px;
    height: 72px;
    margin-left: 35px;
    margin-top: 10px;
    padding: 19px 32.6px 19.4px 18px;
    border-radius: 4px;
    background-color: #e8f0fd;
    .Title {
      width: 500px;
      height: 21px;
      font-family: Inter;
      font-size: 18px;
      font-weight: normal;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.56;
      letter-spacing: normal;
      text-align: center;
      color: #222222;
    }
  }
`;

function Navbar({ dashData }) {
  return (
    <MainDiv data-testid="navbar">
      <Row>
        <span className="Dashboard">
          <T id="Dashboard" />
        </span>
        <Col className="ModifiedCol">
          <img height="30" src={icon} />
        </Col>
      </Row>
      <div className="HeadingDiv">
        <img src={star} />
        <span className="Title">
          {' '}
          <T id="invoices-pay" values={{ dollar: dashData.dollar }} />
        </span>
      </div>
    </MainDiv>
  );
}

Navbar.propTypes = {
  dashData: PropTypes.object
};

export default memo(Navbar);
