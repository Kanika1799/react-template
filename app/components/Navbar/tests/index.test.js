/**
 *
 * Tests for Navbar
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Navbar from '../index';
import mockData from '@services/data.json';
describe('<Navbar />', () => {
  it('should render and match the snapshot', () => {
    const { data } = mockData;
    const { baseElement } = renderWithIntl(<Navbar dashData={data} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Navbar component', () => {
    const { data } = mockData;
    const { getAllByTestId } = renderWithIntl(<Navbar dashData={data} />);
    expect(getAllByTestId('navbar').length).toBe(1);
  });
});
