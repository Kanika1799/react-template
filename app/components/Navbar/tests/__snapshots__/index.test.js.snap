// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Navbar__MainDiv-i5nfb1-0 kIMhYD"
      data-testid="navbar"
    >
      <div
        class="ant-row"
      >
        <span
          class="Dashboard"
        >
          DASHBOARD
        </span>
        <div
          class="ant-col ModifiedCol"
        >
          <img
            height="30"
            src="IMAGE_MOCK"
          />
        </div>
      </div>
      <div
        class="HeadingDiv"
      >
        <img
          src="IMAGE_MOCK"
        />
        <span
          class="Title"
        >
           
          You have saved $345.9 by paying invoices in time. 
        </span>
      </div>
    </div>
  </div>
</body>
`;
